### PR TITLE
Add CPU support and update README

### PR DESCRIPTION
**Pull request by jiayisunx**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,6 +50,8 @@ _(Support for PixArt-Alpha is experimental.)_
 
 You can use the [`prepare_results.py`](./prepare_results.py) script to generate a consolidated CSV file and a plot to visualize the results from it. This is best used after you have run a couple of benchmarking experiments already and have their corresponding CSV files.
 
+The script also supports CPU platforms, you can refer to the [`experiment-scripts/run_sd_cpu.sh`](./experiment-scripts/run_sd_cpu.sh) for some reference experiment commands.
```
Comment (sayakpaul) [inline README.md:55]: Let's also report expected speedup numbers.

File: utils/benchmarking_utils.py [hunk 2]
```diff
@@ -96,10 +96,13 @@ def generate_csv_dict(pipeline_cls: str, args, time: float, memory: float) -> Di
...
         "do_quant": args.do_quant,
         "time (secs)": time,
-        "memory (gbs)": memory,
-        "actual_gpu_memory (gbs)": f"{(TOTAL_GPU_MEMORY):.3f}",
         "tag": args.tag,
     }
+    if args.device == "cuda":
+        memory = bytes_to_giga_bytes(torch.cuda.max_memory_allocated())  # in GBs.
+        TOTAL_GPU_MEMORY = torch.cuda.get_device_properties(0).total_memory / (1024**3)
+        data_dict["memory (gbs)"] = memory,
```
Comment (sayakpaul) [inline utils/benchmarking_utils.py]:
```suggestion
        data_dict["memory (gbs)"] = memory
```

**Review by sayakpaul — COMMENTED** [review]
Thanks for your contributions!

Just two comments.

File: README.md (hunk 1 above)
Comment (jiayisunx) [reply README.md:55]: Sure, expected speedup numbers and experiment results will be shown in the near future.

Comment (sayakpaul) [reply README.md:55]: Then I am afraid I cannot merge the PR without that. We need to have the numbers.

Comment (jiayisunx) [reply README.md:55]: I have added the expected speedup number, please help review again, thanks!

**Review by sayakpaul — APPROVED** [review]

**Comment by sayakpaul** [comment]
4X latency is massive. You don't need any extra setup to achieve this feat, right? If so, it would be nice to document them as thoroughly as possible.

**Comment by jiayisunx** [comment]
> 4X latency is massive. You don't need any extra setup to achieve this feat, right? If so, it would be nice to document them as thoroughly as possible.

Yes, pure, native PyTorch env. The optimization of dynamic int8 quantization has a functionality issue on CPU, which we are still working on. We may show more experiment results after all optimizations can be applied.

**Comment by sayakpaul** [comment]
Okay then we'd want to make that explicitly clear from the README. Otherwise, it's still incomplete in my opinion.

**Comment by jiayisunx** [comment]
> Okay then we'd want to make that explicitly clear from the README. Otherwise, it's still incomplete in my opinion.

Can you be more specific about what you would like me to add to README?

**Comment by sayakpaul** [comment]
The current changes are fine except for we're not specifying what you told me here:
https://github.com/huggingface/diffusion-fast/pull/13#issuecomment-2112234531

So this makes it incomplete.

**Comment by jiayisunx** [comment]
> The current changes are fine except for we're not specifying what you told me here: [#13 (comment)](https://github.com/huggingface/diffusion-fast/pull/13#issuecomment-2112234531)
> 
> So this makes it incomplete.

I have written these optimizations (BFloat16, SDPA, torch.compile, Combining q,k,v projections) can run on CPU platforms, not included the optimization of dynamic int8 quantization here.

**Comment by sayakpaul** [comment]
Sorry that was my oversight. Thanks!

**Pull request merged**